### PR TITLE
removed usage of realpath in cache key

### LINKED ISSUE #1994: Avoid using realpath in Twig_Loader_Filesystem
As of twig 1.18.0 the findTemplate method from the class Twig_Loader_Filesystem uses internally realpath function to cache the actual path to the target template. However the result of findTemplate method is used as the cache key for given template. This cause an error of mismatching the cache filenames when application is built on separate server and then is deployed to backend. In such a case absolute path on backend and build server is different and actual cache generating will be done the second time after deploying when template will be needed for the first time.

### PR DESCRIPTION
**Pull request by fabpot**
closes #1994

### CODE REVIEW

File: lib/Twig/Loader/Filesystem.php [hunk 1]
```diff
@@ -257,4 +253,19 @@ protected function validateName($name)
...
     }
+
+    private function normalizePath($path)
+    {
+        $parts = explode('/', str_replace('\\', '/', $path));
+        $new = array();
+        foreach ($parts as $i => $part) {
+            if ('..' === $part) {
+                array_pop($new);
+            } elseif ('.' !== $part && ('' !== $part || $i == 0)) {
```
Comment (stloyd) [inline lib/Twig/Loader/Filesystem.php]: Should be `$i === 0`.

File: lib/Twig/Loader/Filesystem.php [hunk 2]
```diff
@@ -257,4 +253,19 @@ protected function validateName($name)
...
         }
     }
+
+    private function normalizePath($path)
+    {
+        $parts = explode('/', str_replace('\\', '/', $path));
+        $new = array();
+        foreach ($parts as $i => $part) {
+            if ('..' === $part) {
+                array_pop($new);
```
Comment (stof) [inline lib/Twig/Loader/Filesystem.php]: you should check first that ``$new`` is not empty here

Comment (fabpot) [inline lib/Twig/Loader/Filesystem.php]: It cannot be empty because of the call to `validateName`

File: lib/Twig/Loader/Filesystem.php (hunk 1 above)
Comment (fabpot) [inline lib/Twig/Loader/Filesystem.php]: done

**Pull request merged**